README.md

```diff
@@ -4,132 +4,949 @@
+- **Native Scaling**: Leverage Kubernetes autoscaling, resource limits, and scheduling
+- **Built-in Observability**: Integrate with existing Prometheus, Grafana, and logging infrastructure
 
-The kagent framework is designed to be easy to understand and use, and to provide a flexible and powerful way to build and manage AI agents.
+#### Additional Benefits
+
+1. **Security by Default**: Built-in security best practices with Kubernetes-native access controls
+2. **Operational Excellence**: Comprehensive observability, monitoring, and debugging capabilities
+3. **Extensibility**: Flexible architecture that adapts to your specific requirements
+4. **Production Ready**: Battle-tested in real-world deployments across multiple industries
+
+### Real-World Impact
+
+kagent is already being used in production by organizations across various industries:
+
+- **Telecommunications**: Complex infrastructure monitoring and malicious user detection
+- **Financial Services**: Secure, auditable AI-powered operations with compliance requirements
+- **Identity Verification**: Reliable automation for critical verification platforms
+- **Platform Engineering**: Cloud-native platform solutions for internal teams and customers
+
+### Trusted by Industry Leaders
+
+kagent is trusted by leading organizations worldwide:
+
+<div align="center">
+  <p><strong>Solo.io</strong> • <strong>Amdocs</strong> • <strong>Au10tix</strong> • <strong>Krateo</strong></p>
+  <p><em>Join these organizations in leveraging kagent for production AI agent deployments</em></p>
+</div>
+
+---
+
+## 🏗️ Architecture
+
+kagent is built on a microservices architecture that seamlessly integrates with Kubernetes. The framework consists of four core components that work together to provide a complete AI agent platform.
 
 <div align="center">
-  <img src="img/arch.png" alt="kagent" width="500">
+  <img src="img/arch.png" alt="kagent Architecture" width="700">
+  <p><em>kagent's modular architecture enables flexible and scalable agent deployments</em></p>
+  <p><a href="https://kagent.dev">View detailed architecture documentation and interactive diagrams on kagent.dev</a></p>
 </div>
 
-Kagent has 4 core components:
+### High-Level Architecture Overview
+
+The kagent architecture follows cloud-native principles, with each component designed to be independently scalable, maintainable, and observable. The system leverages Kubernetes Custom Resources (CRDs) for declarative configuration, ensuring that all agent definitions, tool configurations, and model settings are stored as Kubernetes resources.
+
+### Data Flow
+
+1. **User Interaction**: Users interact with agents via the Web UI, CLI, or Kubernetes API
+2. **Controller Processing**: The Controller watches Custom Resources and manages agent lifecycle
+3. **Engine Execution**: The Engine executes agent logic using the configured LLM provider
+4. **Tool Invocation**: Agents invoke tools through MCP servers or direct Kubernetes API calls
```

Comment:
tool calling is always done through the MCP servers (the mcp servers might call k8s apis, clis etc).